File: BackEnd/RestaurantApiProject/Repositories/FoodRepository.cs

using RestaurantApiProject.Contracts;
using RestaurantApiProject.Models;
using RestaurantApiProject.Services;
using System;
using System.Collections.Generic;
using System.Linq;

namespace RestaurantApiProject.Repositories
{
    public class FoodRepository : BaseRepository<Food>, IFoodRepository
    {

        public FoodRepository(RestaurantProjectContext restaurantProjectContext)
            :base(restaurantProjectContext)
        {
        }

        /// <summary>
        /// Get specfic type of food
        /// </summary>
        /// <param name="type">
        /// The wanted type of food
        /// </param>
        /// <returns>
        /// All food that have the same wanted type
        /// </returns>
        public List<Food> getFoodByType(string type)
        {
            try
            {
                List<Food> listOfFood = base.context.Food.Where(s => s.FoodType == type).ToList();

                if (listOfFood == null)
                {
                    return null;
                }

                return listOfFood;
            }
            catch (Exception e)
            {
                throw new Exception(e.Message);

            }
        }

    }
}

File: Baladina_Mobile_App/baladina_mobile_app/lib/data/Repositories/food_repository.dart
import 'package:baladina_website/data/Models/food.dart';
import 'package:baladina_website/data/Services/food_web_service.dart';

class FoodRepository {
  FoodWebService foodWebService;

  FoodRepository(this.foodWebService);

  Future<List<Food>> getAll() async {
    final food = await this.foodWebService.getAll();
    return food.map((food) => Food.fromJson(food)).toList();
  }
}
